src/config/config.ts

```diff
@@ -32,3 +41,34 @@ export const getHttpProviderUri = (networkId: string): string => {
   if (networkId === "homestead") return `https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`;
   return `https://${networkId}.infura.io/v3/${INFURA_PROJECT_ID}`;
 };
+
+export const getEtherscanNetworkApiDetails = (chain: string, chainId: number): EtherscanNetworkApiDetails => {
+  let url = "";
+  let apiKey = "";
+  let networkHostname = "";
+
+  if (chainId === 1 || chainId === 3 || chainId === 4 || chainId === 5 || chainId === 42) {
+    chain = "ETH";
+    if (chainId === 3) {
+      networkHostname = "-ropsten";
+    } else if (chainId === 4) {
+      networkHostname = "-rinkeby";
+    } else if (chainId === 5) {
+      networkHostname = "-goerli";
+    } else if (chainId === 42) {
+      networkHostname = "-kovan";
+    }
+    url = `https://api${networkHostname}.etherscan.io`;
+  } else if (chainId === 137 || chainId === 80001) {
```

Comment:
ah i meant the various `urls` in config here nia, the function itself can:
- remain at where they were use
- if used more than once, in utils

this is fine, we leave it as it is first